File: qhse-frontend/src/app/shared/models/resultat-kpi-response.ts
import { NiveauVariation } from '../enums/niveau-variation.enum';
import { Tendance } from '../enums/tendance.enum';
import { UniteKpi } from '../enums/unite-kpi.enum';

export interface ResultatKpiResponse {
  id: number;
  kpiId: number;
  kpiNom: string;
  kpiUnite: UniteKpi;
  categorieCode: string;
  categorieLibelle: string;
  periodeN1: number;
  periodeN: number;
  valeurN1?: number;
  valeurN?: number;
  variationAbsolue?: number;
  variationRelative?: number;
  niveauVariation?: NiveauVariation;
  tendance?: Tendance;
  analyseIa?: string;
  createdAt?: string;
}

File: QHSEAnalytics/src/main/java/com/QHSEAnalytics/dto/response/ResultatKpiResponse.java
package com.QHSEAnalytics.dto.response;

import com.QHSEAnalytics.enums.NiveauVariation;
import com.QHSEAnalytics.enums.QualityStatus;
import com.QHSEAnalytics.enums.Tendance;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ResultatKpiResponse {
    private Long id;
    private Long kpiId;
    private String kpiNom;
    private String kpiUnite;
    private String categorieCode;
    private String categorieLibelle;
    private Integer periodeN1;
    private Integer periodeN;
    private Double valeurN1;
    private Double valeurN;
    private Double variationAbsolue;
    private Double variationRelative;
    private NiveauVariation niveauVariation;
    private Tendance tendance;
    private Double confidenceScore;
    private QualityStatus qualityStatus;
    private String analyseIa;
    private LocalDateTime createdAt;
}
